src/cli.ts

```diff
@@ -144,43 +144,54 @@ function runWithPaths(paths: Array<string>, options: CLIOptions, callback: ((err
   let errors: Array<Error> = [];
   let pending = paths.slice();
 
-  function processPath(path: string): void {
-    stat(path, (err, info) => {
-      if (err) { errors.push(err); }
-      else if (info.isDirectory()) {
-        processDirectory(path);
-      } else {
-        processFile(path);
-      }
-    });
+  async function processPath(path: string): Promise<void> {
+    const info = await stat(path);
+
+    if (!info) {
+      // Skip
+      return processNext();
+    }
+
+    if (info.isDirectory()) {
+      return processDirectory(path)
+        .then(processNext);
```

Comment:
Let's not mix manual promise syntax with a sync/await. This should be:

```ts
await processDirectory(path);
await processNext();
```

And probably just have an `else` containing `await processFile(path);`.